Fix/25091

leanzafrancesco opened the pull request:
Fixed order totals calculation if it contains taxable and non-taxable products and percentage coupons.

### All Submissions:

* [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Fix the issues mentioned in #25091 
These issues are due to:

- rounding issues when calculating subtotals and coupon taxes; 
- the coupon taxes were calculated also for non-taxable items.

These changes make the order calculate coupon discounts and taxes correctly like in cart.

Closes #25091 .

### How to test the changes in this Pull Request:

To test the behavior you can follow the steps mentioned in #25091 or you can simply use the unit test included for these changes.

### Other information:
This will be the result:
![comparison](https://user-images.githubusercontent.com/16973833/69332402-e9f3a100-0c56-11ea-8e5c-9824b47c6793.png)

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] Have you written new tests for your changes, as applicable?
* [x] Have you successfully run tests with your changes locally?

### Changelog entry

> Fix - order totals calculation if the order contains taxable and non-taxable products and percentage coupons. Added related unit test.

leanzafrancesco committed:
Fixed order totals calculation if it contains taxable and non-taxable products and percentage coupons
Added PHPUnit tests

peterfabian commented:
Hi @leanzafrancesco , thanks for your contribution in this tricky area of tax calculations. I'm having a closer look at this one. Could you please resolve the conflicts as there has been a recent update to the logic there to unify the calculations (which still does not solve the problem you're trying to solve here, but it changed the code you are touching here).

leanzafrancesco committed:
Merge branch 'master' into fix/25091

leanzafrancesco commented:
hi @peterfabian, 
I've just checked and resolved the conflicts (I've also removed the `round_item_subtotal` method I had added since I have just noticed the class is using the same method as static through **WC_Item_Totals**)

Konamiman reviewed (approved):
Code looks good and manual testing worked as expected.
